### PR TITLE
ui: refactor labels with tooltip in forms

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Adds new Vue component TooltipLabel for easier re-use
Moved TooltipButton.vue (and TooltipLabel.vue) to `components/widgets` folder.
Additionally, cosmetic changes to add compute/service offering, network offering and shared network form.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Screenshot from 2021-07-01 13-23-54
Screenshot from 2021-07-01 13-32-42

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by utchoang — APPROVED** [review]
LGTM!

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@blueorangutan ui

File: ui/src/views/compute/CreateKubernetesCluster.vue [hunk 1]
```diff
@@ -253,51 +198,31 @@
           </a-form-item>
           <div v-if="usePrivateRegistry">
             <a-form-item>
-              <span slot="label">
-                {{ $t('label.username') }}
-                <a-tooltip :title="apiParams.dockerregistryusername.description">
-                  <a-icon type="info-circle" style="color: rgba(0,0,0,.45)" />
-                </a-tooltip>
-              </span>
+              <tooltip-label slot="label" :title="$t('label.dockerregistryusername')" :tooltip="apiParams.dockerregistryusername.description"/>
```
Comment (davidjumani) [inline ui/src/views/compute/CreateKubernetesCluster.vue:201]: `label.dockerregistryusername` Needs to be added to en.json

Comment (davidjumani) [reply ui/src/views/compute/CreateKubernetesCluster.vue:201]: ^^ for following labels as well

File: ui/src/views/storage/UploadLocalVolume.vue [hunk 2]
```diff
@@ -103,12 +88,7 @@
           </a-select>
         </a-form-item>
         <a-form-item>
-          <span slot="label">
-            {{ $t('label.volumechecksum') }}
-            <a-tooltip :title="apiParams.checksum.description">
-              <a-icon type="info-circle" style="color: rgba(0,0,0,.45)" />
-            </a-tooltip>
-          </span>
+          <tooltip-label slot="label" :title="$t('label.volumechecksum')" :tooltip="apiParams.volumechecksum.description"/>
```
Comment (Pearl1594) [inline ui/src/views/storage/UploadLocalVolume.vue:91]:
```suggestion
          <tooltip-label slot="label" :title="$t('label.volumechecksum')" :tooltip="apiParams.checksum.description"/>
```

As the form fails to load with the following error
image

File: ui/src/views/storage/CreateVolume.vue [hunk 3]
```diff
@@ -54,12 +44,7 @@
         </a-select>
       </a-form-item>
       <a-form-item>
-        <span slot="label">
-          {{ $t('label.diskoffering') }}
-          <a-tooltip :title="apiParams.diskofferingid.description || 'Disk Offering'">
-            <a-icon type="info-circle" style="color: rgba(0,0,0,.45)" />
-          </a-tooltip>
-        </span>
+        <tooltip-label slot="label" :title="$t('label.diskoffering')" :tooltip="apiParams.diskoffering.description"/>
```
Comment (davidjumani) [inline ui/src/views/storage/CreateVolume.vue:47]:
```suggestion
        <tooltip-label slot="label" :title="$t('label.diskofferingid')" :tooltip="apiParams.diskofferingid.description"/>
```

File: ui/src/views/storage/UploadLocalVolume.vue (hunk 2 above)
Comment (davidjumani) [inline ui/src/views/storage/UploadLocalVolume.vue:91]:
```suggestion
          <tooltip-label slot="label" :title="$t('label.checksum')" :tooltip="apiParams.checksum.description"/>
```

File: ui/src/views/compute/CreateKubernetesCluster.vue (hunk 1 above)
Comment (shwstppr) [reply ui/src/views/compute/CreateKubernetesCluster.vue]: @davidjumani Reverted title to `label.username` as earlier

File: ui/src/views/storage/UploadLocalVolume.vue (hunk 2 above)
Comment (shwstppr) [reply ui/src/views/storage/UploadLocalVolume.vue]: Made it `<tooltip-label slot="label" :title="$t('label.volumechecksum')" :tooltip="apiParams.checksum.description"/>`  with title, tooltip values as they were before refactor

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by davidjumani — APPROVED** [review]

**Review by Pearl1594 — APPROVED** [review]
LGTM

**Review by utchoang — APPROVED** [review]
LGTM!

**Pull request merged**